src/ipc_listener.cpp

```diff
@@ -208,9 +260,13 @@ int zmq::ipc_listener_t::set_address (const char *addr_)
 
 int zmq::ipc_listener_t::close ()
 {
-    zmq_assert (s != retired_fd);
-    int rc = ::close (s);
-    errno_assert (rc == 0);
+    int rc;
+
+    if ( s != -1 ) {
+      zmq_assert (s != retired_fd);
```

Comment:
On Mon, Mar 14, 2016 at 11:10 AM, Luca Boccassi
<notifications@github.com> wrote:
> If you need more time, I can merge as-is for now, and then you can send
> another PR to fix it when it's ready. Or maybe it's easier for you to work
> on the same branch?

Actually, the cleanup was far easier than I thought.  I reverted the
goto's back to returning -1, but also cleaning up any created
directory.  I also reverted the API checks at the beginning of
ipc_listener_t::close() to the prior checks.  I also added code to
preserve errno when trying to cleanup in these cases.

I also have the formatting cleaned up, and tweaked my code to have
opening braces on the same line (to match the rest of the file).

Pete